src/value/mod.rs

```diff
@@ -487,11 +487,37 @@ pub trait Struct {
 
 pub trait Builder {
     type Element: Element + ?Sized;
+    type SymbolToken: SymbolToken + ?Sized;
     type Sequence: Sequence<Element = Self::Element> + ?Sized;
+    type Struct: Struct<FieldName = Self::SymbolToken, Element = Self::Element> + ?Sized;
 
     /// Build a `null` from IonType using Builder
     fn new_null(e_type: IonType) -> Self::Element;
 
+    /// Build a `bool` using Builder
+    fn new_bool(bool: bool) -> Self::Element;
+
+    /// Build a `string` using Builder
+    fn new_string(str: &'static str) -> Self::Element;
+
+    /// Build a `symbol` from SymbolToken using Builder
+    fn new_symbol(sym: Self::SymbolToken) -> Self::Element;
+
+    /// Build a `i64` using Builder
+    fn new_i64(int: i64) -> Self::Element;
+
+    /// Build a `big int` using Builder
+    fn new_big_int(big_int: BigInt) -> Self::Element;
+
+    /// Build a `decimal` using Builder
+    fn new_decimal(decimal: Decimal) -> Self::Element;
+
+    /// Build a `timestamp` using Builder
+    fn new_timestamp(timestamp: Timestamp) -> Self::Element;
+
+    /// Build a `f64` using Builder
+    fn new_f64(float: f64) -> Self::Element;
+
     /// Build a `clob` using Builder
     fn new_clob(bytes: &'static [u8]) -> Self::Element;
 
```

Comment:
I somehow missed this in an earlier commit--this should not be `'static` for sure, if we cannot have an appropriate lifetime due to GAT, then let's just use the owned type (e.g. `Vec<u8>`)